ru/lessons/libraries/benchee.md

```diff
@@ -0,0 +1,326 @@
+* **console** &mdash; ассоциативный массив или ключевой список со следующими опциями-атомами в качестве ключей и произвольными значениями. Доступные опции:
+  * **comparison** &mdash; показывать ли сравнение результатов разных тестов (*х* в столько-то раз медленее, чем *y*). По умолчанию `true`, но можно отключить при помощи `false`.
+  * **unit_scaling** &mdash; масштабирование &mdash; стратегия выбора единиц измерения для времени и количества. Когда масштабирование включено, `Benchee` подыщет наиболее подходящее отображение для чисел. Например, `1_200_000` масштабируется в 1,2 М, а `800_000` в 800 К. Стратегия масштабирования определяет то, как `Benchee` будет выбирать отображение числовых значений в случаях, когда "наилучший" вид разный для разных чисел. Всего есть четыре стратегии, у каждой &mdash; имя-атом. По умолчанию используется `:best`:
+    * **best** &mdash; будет выбрано наиболее часто встречающееся сокращение. В случае совпадения, будет выбрано наибольшее.
+    * **largest** &mdash; среди наилучших сокращений будет выбрано наибольшее
+    * **smallest** &mdash; среди наилучших будет выбрано наименьшее
+    * **none** &mdash; масштабирование не будет использоваться вовсе. Время будет выводится в микросекундах, а количество `ips` &mdash; без тысячных сокращений.
+
+# Наборы параметров
+
+Очень важно тестировать функции с использованием данных, похожих на те, с которыми она скорее всего столкнётся в реальном мире. Часто функция может вести себя по-разному на маленьких и больших наборах данных. Специально для таких случаев `Benchee` поддерживает наборы параметров (`inputs`)! Они позволяют нам тестировать одну и ту же функцию со столькими наборами входных параметров, сколько нам понадобится, а затем отобразить их в результатах.
+
+Посмотрим на наш первый пример:
+
+```elixir
+list = Enum.to_list(1..10_000)
+map_fun = fn(i) -> [i, i * i] end
+
+Benchee.run(%{
+  "flat_map"    => fn -> Enum.flat_map(list, map_fun) end,
+  "map.flatten" => fn -> list |> Enum.map(map_fun) |> List.flatten end
+})
+```
+
+В нём мы используем один-единственный список целых чисел от 1 до 10 000. Давайте вместо него используем несколько разных входных параметров, чтобы можно было сравнить результаты на малых и больших списках. Должно получиться так:
+
+```elixir
+map_fun = fn(i) -> [i, i * i] end
+inputs = %{
+  "small list" => Enum.to_list(1..100),
+  "medium list" => Enum.to_list(1..10_000),
+  "large list" => Enum.to_list(1..1_000_000)
+}
+
+Benchee.run(%{
+  "flat_map"    => fn(list) -> Enum.flat_map(list, map_fun) end,
+  "map.flatten" => fn(list) -> list |> Enum.map(map_fun) |> List.flatten end
+}, inputs: inputs)
+```
+
+Можно заметить два существенных различия. Во-первых, мы создали ассоциативный массив `inputs`, в котором хранится информация о наборах входных параметров. Во-вторых, мы передаём его в виде настройки в `Benchee.run/2`.
+
+И так как теперь наша функция должна уметь принимать входной аргумент, мы должны заменить 
```

Comment:
This sentence is followed by a code block, I suppose that it should end with a colon.